[GraalVM for JDK 24+] All native integration tests failing

### Describe the bug

The last good revision is: ddf332b280fd8f91bd1257c9c3962d9ecb9874f7 (which got built in our CI yesterday and was OK). Now - starting with revision ddd0ebc55aa65f1eba88cd7cb0e7def26a467903 - all integration tests with Mandrel for JDK 24 (e.g. ` mandrel-java24-24.2.1.0-Final`) fail at runtime with:

```
Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 8.673 s <<< FAILURE! -- in io.quarkus.it.jpa.derby.JPAFunctionalityInGraalITCase
Error:  io.quarkus.it.jpa.derby.JPAFunctionalityInGraalITCase.testJPAFunctionalityFromServlet -- Time elapsed: 1.008 s <<< ERROR!
org.apache.http.NoHttpResponseException: localhost:8081 failed to respond
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:141)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:56)
	at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:259)
	at org.apache.http.impl.AbstractHttpClientConnection.receiveResponseHeader(AbstractHttpClientConnection.java:294)
	at org.apache.http.impl.conn.DefaultClientConnection.receiveResponseHeader(DefaultClientConnection.java:257)
	at org.apache.http.impl.conn.ManagedClientConnectionImpl.receiveResponseHeader(ManagedClientConnectionImpl.java:207)
	at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:273)
	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:125)
	at org.apache.http.impl.client.DefaultRequestDirector.tryExecute(DefaultRequestDirector.java:679)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:481)
	at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:835)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at io.restassured.internal.RestAssuredHttpBuilder.doRequest(RestAssuredHttpBuilder.java:141)
	at io.restassured.internal.http.HTTPBuilder.doRequest(HTTPBuilder.java:496)
	at io.restassured.internal.http.HTTPBuilder.request(HTTPBuilder.java:453)
	at io.restassured.internal.http.HTTPBuilder$request$2.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:166)
	at io.restassured.internal.RequestSpecificationImpl.sendHttpRequest(RequestSpecificationImpl.groovy:1480)
	at io.restassured.internal.RequestSpecificationImpl.this$2$sendHttpRequest(RequestSpecificationImpl.groovy)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:343)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:328)
	at groovy.lang.MetaClassImpl.doInvokeMethod(MetaClassImpl.java:1333)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1088)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1007)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:790)
	at io.restassured.internal.RequestSpecificationImpl.invokeMethod(RequestSpecificationImpl.groovy)
	at org.codehaus.groovy.runtime.callsite.PogoInterceptableSite.call(PogoInterceptableSite.java:46)
	at org.codehaus.groovy.runtime.callsite.PogoInterceptableSite.callCurrent(PogoInterceptableSite.java:57)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
	at io.restassured.internal.RequestSpecificationImpl.sendRequest(RequestSpecificationImpl.groovy:1229)
	at io.restassured.internal.RequestSpecificationImpl.this$2$sendRequest(RequestSpecificationImpl.groovy)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:343)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:328)
	at groovy.lang.MetaClassImpl.doInvokeMethod(MetaClassImpl.java:1333)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1088)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1007)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:790)
	at io.restassured.internal.RequestSpecificationImpl.invokeMethod(RequestSpecificationImpl.groovy)
	at org.codehaus.groovy.runtime.callsite.PogoInterceptableSite.call(PogoInterceptableSite.java:46)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:166)
	at io.restassured.internal.filter.SendRequestFilter.filter(SendRequestFilter.groovy:30)
	at io.restassured.filter.Filter$filter.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at io.restassured.filter.Filter$filter$0.call(Unknown Source)
	at io.restassured.internal.filter.FilterContextImpl.next(FilterContextImpl.groovy:72)
	at io.restassured.filter.time.TimingFilter.filter(TimingFilter.java:56)
	at io.restassured.filter.Filter$filter$0.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at io.restassured.filter.Filter$filter$0.call(Unknown Source)
	at io.restassured.internal.filter.FilterContextImpl.next(FilterContextImpl.groovy:72)
	at io.restassured.filter.log.StatusCodeBasedLoggingFilter.filter(StatusCodeBasedLoggingFilter.java:112)
	at io.restassured.filter.log.ResponseLoggingFilter.filter(ResponseLoggingFilter.java:31)
	at io.restassured.filter.Filter$filter$0.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at io.restassured.filter.Filter$filter$0.call(Unknown Source)
	at io.restassured.internal.filter.FilterContextImpl.next(FilterContextImpl.groovy:72)
	at io.restassured.filter.log.RequestLoggingFilter.filter(RequestLoggingFilter.java:148)
	at io.restassured.filter.Filter$filter$0.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at io.restassured.filter.Filter$filter.call(Unknown Source)
	at io.restassured.internal.filter.FilterContextImpl.next(FilterContextImpl.groovy:72)
	at io.restassured.filter.FilterContext$next.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:148)
	at io.restassured.internal.filter.CsrfFilter.filter(CsrfFilter.groovy:70)
	at io.restassured.filter.Filter$filter.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:157)
	at io.restassured.internal.filter.FilterContextImpl.next(FilterContextImpl.groovy:72)
	at io.restassured.filter.FilterContext$next.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:148)
	at io.restassured.internal.RequestSpecificationImpl.applyPathParamsAndSendRequest(RequestSpecificationImpl.groovy:1704)
	at io.restassured.internal.RequestSpecificationImpl.this$2$applyPathParamsAndSendRequest(RequestSpecificationImpl.groovy)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:343)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:328)
	at groovy.lang.MetaClassImpl.doInvokeMethod(MetaClassImpl.java:1333)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1088)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1007)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:790)
	at io.restassured.internal.RequestSpecificationImpl.invokeMethod(RequestSpecificationImpl.groovy)
	at org.codehaus.groovy.runtime.callsite.PogoInterceptableSite.call(PogoInterceptableSite.java:46)
	at org.codehaus.groovy.runtime.callsite.PogoInterceptableSite.callCurrent(PogoInterceptableSite.java:57)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:203)
	at io.restassured.internal.RequestSpecificationImpl.applyPathParamsAndSendRequest(RequestSpecificationImpl.groovy:1710)
	at io.restassured.internal.RequestSpecificationImpl.this$2$applyPathParamsAndSendRequest(RequestSpecificationImpl.groovy)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:343)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:328)
	at groovy.lang.MetaClassImpl.doInvokeMethod(MetaClassImpl.java:1333)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1088)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1007)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:790)
	at io.restassured.internal.RequestSpecificationImpl.invokeMethod(RequestSpecificationImpl.groovy)
	at org.codehaus.groovy.runtime.callsite.PogoInterceptableSite.call(PogoInterceptableSite.java:46)
	at org.codehaus.groovy.runtime.callsite.PogoInterceptableSite.callCurrent(PogoInterceptableSite.java:57)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:49)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:203)
	at io.restassured.internal.RequestSpecificationImpl.get(RequestSpecificationImpl.groovy:172)
	at io.restassured.internal.RequestSpecificationImpl.get(RequestSpecificationImpl.groovy)
	at io.quarkus.it.jpa.derby.JPAFunctionalityTest.testJPAFunctionalityFromServlet(JPAFunctionalityTest.java:18)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestMethod(QuarkusTestExtension.java:836)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

```

To reproduce: Run a native integration test with Mandrel for JDK 24 (for example `jpa`).

See: https://github.com/graalvm/mandrel/actions/runs/15058004317/job/42328811735#step:12:496

All IT tests are red here (affects GraalVM CE build as well as Mandrel for JDK 24+). Mandrel for JDK 21 is OK.

Here is the exact stack trace:

```
Caused by: java.lang.IllegalStateException: Invalid JKS key store configuration for certificate '<default>'
        at io.quarkus.tls.runtime.keystores.JKSKeyStores.toOptions(JKSKeyStores.java:78)
        at io.quarkus.tls.runtime.keystores.JKSKeyStores.verifyJKSKeyStore(JKSKeyStores.java:34)
        at io.quarkus.tls.runtime.CertificateRecorder.getKeyStore(CertificateRecorder.java:140)
        at io.quarkus.tls.runtime.CertificateRecorder.verifyCertificateConfigInternal(CertificateRecorder.java:103)
        at io.quarkus.tls.runtime.CertificateRecorder.verifyCertificateConfig(CertificateRecorder.java:89)
        at io.quarkus.tls.runtime.CertificateRecorder.validateCertificates(CertificateRecorder.java:57)
        at io.quarkus.runner.recorded.CertificatesProcessor$initializeCertificate426175759.deploy_0(Unknown Source)
        at io.quarkus.runner.recorded.CertificatesProcessor$initializeCertificate426175759.deploy(Unknown Source)
        ... 7 more
Caused by: java.lang.NullPointerException
        at io.netty.util.internal.PlatformDependent0.safeConstructPutInt(PlatformDependent0.java:765)
        at io.netty.util.internal.PlatformDependent.safeConstructPutInt(PlatformDependent.java:556)
        at io.netty.util.internal.ReferenceCountUpdater.setInitialValue(ReferenceCountUpdater.java:67)
        at io.netty.buffer.AbstractReferenceCountedByteBuf.<init>(AbstractReferenceCountedByteBuf.java:50)
        at io.netty.buffer.UnpooledHeapByteBuf.<init>(UnpooledHeapByteBuf.java:51)
        at io.vertx.core.buffer.impl.VertxHeapByteBuf.<init>(VertxHeapByteBuf.java:24)
        at io.vertx.core.buffer.impl.VertxByteBufAllocator$2.newHeapBuffer(VertxByteBufAllocator.java:44)
        at io.netty.buffer.AbstractByteBufAllocator.heapBuffer(AbstractByteBufAllocator.java:169)
        at io.vertx.core.buffer.impl.BufferImpl.<init>(BufferImpl.java:76)
        at io.vertx.core.buffer.impl.BufferImpl.buffer(BufferImpl.java:58)
        at io.vertx.core.buffer.Buffer.buffer(Buffer.java:94)
        at io.quarkus.tls.runtime.keystores.JKSKeyStores.toOptions(JKSKeyStores.java:59)
        ... 14 more
```

We cannot initialize buffers anymore.

> Here is the exact stack trace:
> 
> ```
> Caused by: java.lang.IllegalStateException: Invalid JKS key store configuration for certificate '<default>'
>         at io.quarkus.tls.runtime.keystores.JKSKeyStores.toOptions(JKSKeyStores.java:78)
>         at io.quarkus.tls.runtime.keystores.JKSKeyStores.verifyJKSKeyStore(JKSKeyStores.java:34)
>         at io.quarkus.tls.runtime.CertificateRecorder.getKeyStore(CertificateRecorder.java:140)
>         at io.quarkus.tls.runtime.CertificateRecorder.verifyCertificateConfigInternal(CertificateRecorder.java:103)
>         at io.quarkus.tls.runtime.CertificateRecorder.verifyCertificateConfig(CertificateRecorder.java:89)
>         at io.quarkus.tls.runtime.CertificateRecorder.validateCertificates(CertificateRecorder.java:57)
>         at io.quarkus.runner.recorded.CertificatesProcessor$initializeCertificate426175759.deploy_0(Unknown Source)
>         at io.quarkus.runner.recorded.CertificatesProcessor$initializeCertificate426175759.deploy(Unknown Source)
>         ... 7 more
> Caused by: java.lang.NullPointerException
>         at io.netty.util.internal.PlatformDependent0.safeConstructPutInt(PlatformDependent0.java:765)
>         at io.netty.util.internal.PlatformDependent.safeConstructPutInt(PlatformDependent.java:556)
>         at io.netty.util.internal.ReferenceCountUpdater.setInitialValue(ReferenceCountUpdater.java:67)
>         at io.netty.buffer.AbstractReferenceCountedByteBuf.<init>(AbstractReferenceCountedByteBuf.java:50)
>         at io.netty.buffer.UnpooledHeapByteBuf.<init>(UnpooledHeapByteBuf.java:51)
>         at io.vertx.core.buffer.impl.VertxHeapByteBuf.<init>(VertxHeapByteBuf.java:24)
>         at io.vertx.core.buffer.impl.VertxByteBufAllocator$2.newHeapBuffer(VertxByteBufAllocator.java:44)
>         at io.netty.buffer.AbstractByteBufAllocator.heapBuffer(AbstractByteBufAllocator.java:169)
>         at io.vertx.core.buffer.impl.BufferImpl.<init>(BufferImpl.java:76)
>         at io.vertx.core.buffer.impl.BufferImpl.buffer(BufferImpl.java:58)
>         at io.vertx.core.buffer.Buffer.buffer(Buffer.java:94)
>         at io.quarkus.tls.runtime.keystores.JKSKeyStores.toOptions(JKSKeyStores.java:59)
>         ... 14 more
> ```
> 
> We cannot initialize buffers anymore.

I tried to trace the issue using the stacktrace but I could not find the reason why it fails.

>         at io.netty.util.internal.PlatformDependent.safeConstructPutInt(PlatformDependent.java:556)
>         at io.netty.util.internal.ReferenceCountUpdater.setInitialValue(ReferenceCountUpdater.java:67)

[ReferenceCountUpdater.java](https://github.com/netty/netty/blob/8c9155cef8805cc34862817b60befa948cf6f1a5/common/src/main/java/io/netty/util/internal/ReferenceCountUpdater.java#L62-L69):
```
    public void setInitialValue(T instance) {
        final long offset = unsafeOffset();
        if (offset == -1) {
            updater().set(instance, initialValue());
        } else {
67          PlatformDependent.safeConstructPutInt(instance, offset, initialValue()); // <------
        }
    }
```

`PlatformDependent.safeConstructPutInt()`  should not be called here as `offset` should be equal to `-1` (which means no Unsafe).
`unsafeOffset()` returns `ReferenceCountUpdater.getUnsafeOffset(AbstractReferenceCountedByteBuf.class, "refCnt");`:

[ReferenceCountUpdater.java](https://github.com/netty/netty/blob/8c9155cef8805cc34862817b60befa948cf6f1a5/common/src/main/java/io/netty/util/internal/ReferenceCountUpdater.java#L43-L52):

```
    public static long getUnsafeOffset(Class<? extends ReferenceCounted> clz, String fieldName) {
        try {
            if (PlatformDependent.hasUnsafe()) {
                return PlatformDependent.objectFieldOffset(clz.getDeclaredField(fieldName));
            }
        } catch (Throwable ignore) {
            // fall-back
        }
        return -1;
    }
```

[PlatformDependent.java](https://github.com/netty/netty/blob/netty-4.1.121.Final/common/src/main/java/io/netty/util/internal/PlatformDependent.java):

```
    private static final Throwable UNSAFE_UNAVAILABILITY_CAUSE = unsafeUnavailabilityCause0();
    
    // ...
    
    public static boolean hasUnsafe() {
        return UNSAFE_UNAVAILABILITY_CAUSE == null;
    }
    
    // ...
    
    private static Throwable unsafeUnavailabilityCause0() {
        // ...
        
        Throwable cause = PlatformDependent0.getUnsafeUnavailabilityCause();
        if (cause != null) {
            return cause;
        }
        
        try {
            boolean hasUnsafe = PlatformDependent0.hasUnsafe();
            logger.debug("sun.misc.Unsafe: {}", hasUnsafe ? "available" : "unavailable");
            return hasUnsafe ? null : PlatformDependent0.getUnsafeUnavailabilityCause();
        } catch (Throwable t) {
            // ...
        }
    }
```

[PlatformDependent0.java](https://github.com/netty/netty/blob/netty-4.1.121.Final/common/src/main/java/io/netty/util/internal/PlatformDependent0.java):

```
    private static final Throwable EXPLICIT_NO_UNSAFE_CAUSE = explicitNoUnsafeCause0();
    private static final Throwable UNSAFE_UNAVAILABILITY_CAUSE;
    static final Unsafe UNSAFE;
    
    // ...
    
    static {
        // ...
        
        Throwable unsafeUnavailabilityCause = null;
        Unsafe unsafe;
        
        // ...
        
        if ((unsafeUnavailabilityCause = EXPLICIT_NO_UNSAFE_CAUSE) != null) {
            // ...
            unsafe = null;
        } else {
            // ...
        }
        
        UNSAFE_UNAVAILABILITY_CAUSE = unsafeUnavailabilityCause;
        UNSAFE = unsafe;
    }
    
    // ...
    
    private static Throwable explicitNoUnsafeCause0() {
        boolean explicitProperty = SystemPropertyUtil.contains("io.netty.noUnsafe");
        boolean noUnsafe = SystemPropertyUtil.getBoolean("io.netty.noUnsafe", false);

        String reason = "io.netty.noUnsafe";
        String unspecified = "<unspecified>";
        String unsafeMemoryAccess = SystemPropertyUtil.get("sun.misc.unsafe.memory.access", unspecified);
        if (!explicitProperty && unspecified.equals(unsafeMemoryAccess) && javaVersion() >= 24) {
            reason = "io.netty.noUnsafe=true by default on Java 24+";
            noUnsafe = true;
        }
        // ...
        
        if (noUnsafe) {
            String msg = "sun.misc.Unsafe: unavailable (" + reason + ')';
            logger.debug(msg);
            return new UnsupportedOperationException(msg);
        }
        // ...
    }
    
    // ...
    
    static Throwable getUnsafeUnavailabilityCause() {
        return UNSAFE_UNAVAILABILITY_CAUSE;
    }
    
    static long objectFieldOffset(Field field) {
        return UNSAFE.objectFieldOffset(field);
    }
    
```

The Netty code looks right to me and it should have called `updater().set(instance, initialValue());` instead.

Is it possible that this has to do with the usage of `addRuntimeReinitializedClass` in `NettyProcessor`?

https://github.com/quarkusio/quarkus/blob/27289e0642b6ccbf92111aab621692bc663d60ec/extensions/netty/deployment/src/main/java/io/quarkus/netty/deployment/NettyProcessor.java#L231-L249

--- a/extensions/netty/deployment/src/main/java/io/quarkus/netty/deployment/NettyProcessor.java
+++ b/extensions/netty/deployment/src/main/java/io/quarkus/netty/deployment/NettyProcessor.java
@@ -246,7 +246,11 @@ class NettyProcessor {
                 // - io.netty.tryUnsafe
                 // - org.jboss.netty.tryUnsafe
                 // - io.netty.tryReflectionSetAccessible
-                .addRuntimeReinitializedClass("io.netty.util.internal.PlatformDependent0");
+                .addRuntimeReinitializedClass("io.netty.util.internal.PlatformDependent0")
+                // Runtime initialize classes to allow netty to use the field offset for testing if unsafe is available or not
+                // See https://github.com/quarkusio/quarkus/issues/47903#issuecomment-2890924970
+                .addRuntimeReinitializedClass("io.netty.util.AbstractReferenceCounted")
+                .addRuntimeReinitializedClass("io.netty.buffer.AbstractReferenceCountedByteBuf");
 
         if (QuarkusClassLoader.isClassPresentAtRuntime("io.netty.buffer.UnpooledByteBufAllocator")) {
             // Runtime initialize due to the use of the io.netty.util.internal.PlatformDependent class

--- a/extensions/netty/runtime/src/main/java/io/quarkus/netty/runtime/graal/NettySubstitutions.java
+++ b/extensions/netty/runtime/src/main/java/io/quarkus/netty/runtime/graal/NettySubstitutions.java
@@ -407,22 +407,6 @@ final class Target_io_netty_channel_nio_NioEventLoop {
     }
 }
 
-@TargetClass(className = "io.netty.buffer.AbstractReferenceCountedByteBuf")
-final class Target_io_netty_buffer_AbstractReferenceCountedByteBuf {
-
-    @Alias
-    @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.FieldOffset, name = "refCnt")
-    private static long REFCNT_FIELD_OFFSET;
-}
-
-@TargetClass(className = "io.netty.util.AbstractReferenceCounted")
-final class Target_io_netty_util_AbstractReferenceCounted {
-
-    @Alias
-    @RecomputeFieldValue(kind = RecomputeFieldValue.Kind.FieldOffset, name = "refCnt")
-    private static long REFCNT_FIELD_OFFSET;
-}
-
 // This class is runtime-initialized by NettyProcessor
 final class Holder_io_netty_util_concurrent_ScheduledFutureTask {
     static final long START_TIME = System.nanoTime();

